app/Console/Commands/ChatChannelSetLastMessageId.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace App\Console\Commands;
+
+use App\Models\Chat\Channel;
+use App\Models\Chat\Message;
+use Illuminate\Console\Command;
+
+class ChatChannelSetLastMessageId extends Command
+{
+    const MAX_PER_LOOP = 1000;
+    const NOTIFICATION_ID_BUFFER = 1000;
+
+    protected $signature = 'chat:channel-set-last-message-id {--chunk-size=1000} {--delay=1}';
+
+    protected $description = 'Updates channels where last_message_id is not set.';
+
+    public function handle()
+    {
+        $delay = $this->getOptionValue('delay', 1);
+        $chunkSize = $this->getOptionValue('chunk-size', 1000);
+        if ($chunkSize < 1) {
+            return;
+        }
+
+        $this->line("Updating chat channels without last_message_id with {$delay}s delay between chunks of {$chunkSize}...");
+
+        $query = Channel::where('last_message_id', null);
+        $progress = $this->output->createProgressBar($query->count());
```

Comment:
not sure about counting unindexed column. At least it currently only takes 800ms so maybe not too bad if only meant to be run once...